### ISSUE TITLE
Error identifying disc.

### ISSUE THREAD

**Issue by abaj8494** [asartori/mopidy-cd#19]
Output of `journalctl -u mopidy -f` is as follows
```
Feb 06 16:49:11 alarmpi mopidy[6105]:   element cdparanoiasrc2: GstMessageError, gerror=(GError)NULL, debug=(string)"../gst-plugins-base/ext/...
Feb 06 16:49:11 alarmpi mopidy[6105]: INFO     2022-02-06 16:49:11,387 [6105:CdBackend-8 (_actor_loop)] mopidy_cd.cdrom
Feb 06 16:49:11 alarmpi mopidy[6105]:   Error identifying disc: gst-resource-error-quark: Could not open CD device for reading. (5)
```
I have no idea how to fix this. Is the problem with cdparanoia, mopidy or gstreamer?

**Comment by abaj8494** [comment]
Okay so, to fix this I had to add the `mopidy` user to the `optical` group using `usermod -aG optical mopidy`

**Comment by abaj8494** [comment]
However, the CdBackend is now raising a `TypeError` as such. 
```
Feb 10 13:47:26 alarmpi mopidy[11413]:   Request from [::ffff:192.168.1.153]:53941: add "CD"
Feb 10 13:47:27 alarmpi mopidy[11413]: TRACE    2022-02-10 13:47:27,714 [11413:CdBackend-8 (_actor_loop)] mopidy.audio.scan
Feb 10 13:47:27 alarmpi mopidy[11413]:   element cdparanoiasrc0: GstMessageDurationChanged;
Feb 10 13:47:27 alarmpi mopidy[11413]: TRACE    2022-02-10 13:47:27,718 [11413:CdBackend-8 (_actor_loop)] mopidy.audio.scan
Feb 10 13:47:27 alarmpi mopidy[11413]:   element typefindelement0: have-type, caps=(structure)"audio/x-raw\,\ format\=\(string\)S16LE\,\ layout\...
Feb 10 13:47:27 alarmpi mopidy[11413]: TRACE    2022-02-10 13:47:27,727 [11413:CdBackend-8 (_actor_loop)] mopidy.audio.scan
Feb 10 13:47:27 alarmpi mopidy[11413]:   element decodebin0: have-audio;
Feb 10 13:47:27 alarmpi mopidy[11413]: TRACE    2022-02-10 13:47:27,729 [11413:CdBackend-8 (_actor_loop)] mopidy.audio.scan
Feb 10 13:47:27 alarmpi mopidy[11413]:   element fakesink0: GstMessageTag, taglist=(taglist)"taglist\,\ discid\=\(string\)77122b09\,\ dis...
Feb 10 13:47:27 alarmpi mopidy[11413]: TRACE    2022-02-10 13:47:27,735 [11413:CdBackend-8 (_actor_loop)] mopidy.audio.scan
Feb 10 13:47:27 alarmpi mopidy[11413]:   element cdparanoiasrc0: GstMessageDurationChanged;
Feb 10 13:47:28 alarmpi mopidy[11413]: DEBUG    2022-02-10 13:47:28,418 [11413:CdBackend-8 (_actor_loop)] mopidy_cd.cdrom
Feb 10 13:47:28 alarmpi mopidy[11413]:   Read disc: MusicBrainz DiscID HVoRVJ8Y5gixR7c8qBjKDsAgqgk-, 9 tracks
Feb 10 13:47:28 alarmpi mopidy[11413]: DEBUG    2022-02-10 13:47:28,421 [11413:CdBackend-8 (_actor_loop)] musicbrainzngs
Feb 10 13:47:28 alarmpi mopidy[11413]:   GET request for http://musicbrainz.org/ws/2/discid/HVoRVJ8Y5gixR7c8qBjKDsAgqgk-?cdstubs=no&inc=artist-credits+recordings&toc=1+9+349003+183+33095+54405+101535+163810+248785+274073+304105+335830
Feb 10 13:47:28 alarmpi mopidy[11413]: DEBUG    2022-02-10 13:47:28,435 [11413:CdBackend-8 (_actor_loop)] musicbrainzngs
Feb 10 13:47:28 alarmpi mopidy[11413]:   requesting with UA Mopidy-Cd/0.5 python-musicbrainzngs/0.6
Feb 10 13:47:29 alarmpi mopidy[11413]: INFO     2022-02-10 13:47:29,137 [11413:CdBackend-8 (_actor_loop)] mopidy_cd.cdrom
Feb 10 13:47:29 alarmpi mopidy[11413]:   Error accessing MusicBrainz: list index out of range
Feb 10 13:47:29 alarmpi mopidy[11413]: INFO     2022-02-10 13:47:29,156 [11413:CdBackend-8 (_actor_loop)] pykka
Feb 10 13:47:29 alarmpi mopidy[11413]:   Exception returned from CdBackend (urn:uuid:5b6c2eaa-e58c-47d9-9c22-8d36ff30fb4a) to caller:
Feb 10 13:47:29 alarmpi mopidy[11413]: Traceback (most recent call last):
Feb 10 13:47:29 alarmpi mopidy[11413]:   File "/usr/lib/python3.10/site-packages/pykka/_actor.py", line 186, in _actor_loop
Feb 10 13:47:29 alarmpi mopidy[11413]:     response = self._handle_receive(envelope.message)
Feb 10 13:47:29 alarmpi mopidy[11413]:   File "/usr/lib/python3.10/site-packages/pykka/_actor.py", line 286, in _handle_receive
Feb 10 13:47:29 alarmpi mopidy[11413]:     return callee(*message.args, **message.kwargs)
Feb 10 13:47:29 alarmpi mopidy[11413]:   File "/usr/lib/python3.10/site-packages/mopidy_cd/backend.py", line 44, in lookup
Feb 10 13:47:29 alarmpi mopidy[11413]:     album = CdLibrary._make_album(disc)
Feb 10 13:47:29 alarmpi mopidy[11413]:   File "/usr/lib/python3.10/site-packages/mopidy_cd/backend.py", line 114, in _make_album
Feb 10 13:47:29 alarmpi mopidy[11413]:     return Album(
Feb 10 13:47:29 alarmpi mopidy[11413]:   File "/usr/lib/python3.10/site-packages/mopidy/models/immutable.py", line 159, in __call__
Feb 10 13:47:29 alarmpi mopidy[11413]:     instance = super().__call__(*args, **kwargs)
Feb 10 13:47:29 alarmpi mopidy[11413]:   File "/usr/lib/python3.10/site-packages/mopidy/models/immutable.py", line 32, in __init__
Feb 10 13:47:29 alarmpi mopidy[11413]:     raise TypeError(
Feb 10 13:47:29 alarmpi mopidy[11413]: TypeError: __init__() got an unexpected keyword argument 'images'
```
I think this may be related to the tracks on the audioCD being labeled as track1, track2, etc, and MusicBrainz being unable to retrieve the appropriate metadata.

**Comment by abaj8494** [comment]
Okay so I managed to fix this by installing from source. Ran the standard `git clone https://github.com/antosart/mopidy-cd.git`, `cd`'d into mopidy-cd and ran `sudo python setup.py install` (the `sudo` is important). From there I fixed a couple python backend errors by replacing the line `from collections import Mapping, namedtuple` to `from collections.abc import Mapping` and `from collections import import namedtuple`.

Restarted the mopidy service with `sudo systemctl restart mopidy` and my Mahler disc is now playing 🥳️🎉️🎶️!
